pkgs/servers/http/merecat/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
+, fetchzip
+, pkg-config
+, libconfuse
+}:
+
+stdenv.mkDerivation rec {
+  pname = "merecat";
+  version = "2.31";
+
+  src = fetchzip {
+    url = "ftp://ftp.troglobit.com/merecat/merecat-${version}.tar.xz";
```

Comment:
It's only way faster in relative terms (and excluding build time). In absolute terms it's only a few seconds more. I prefer to use the release tarball linked on the homepage since it's more likely to be stable, but will change it if those few seconds are an actual concern or if another maintainer weighs in

Reply:
The official release is also linked here https://github.com/troglobit/merecat/releases/tag/v2.31

But another question is - is there a reason to pull the package specifically from the url instead of `fetchFromGitHub`?

Reply:
> is there a reason to pull the package specifically from the url instead of fetchFromGitHub? It's a bit safer given their project site is down right now (ftp host isn't, but still, not a great sign for reliability).

Aren't release tarballs preferred over repos when available? If it goes down it can be changed later

Reply:
> Aren't release tarballs preferred over repos when available?

Kind of. The specific fragment is here: https://nixos.org/manual/nixpkgs/stable/#sec-sources
`fetchFromGitHub` does the right thing and it doesn't actually clone the repo. It handles caching well too.

In this case I'd really rather use the github source, because:
- the project itself seems semi-abandonned with the site not working
- the only download link I see officially referred to (https://github.com/troglobit/merecat/blob/master/Dockerfile#L12) doesn't actually work
- ftp with multiple connections is still occasionally problematic behind corporate proxies

I'd rather follow:
> The general guideline is that you should package reproducible sources with a high degree of availability.
And point at github with better changes of longterm survival.

Reply:
> the project itself seems semi-abandonned with the site not working

It isn't, I recently contacted the author

Anyway, I checked the repo and tarball with nix-diff, and since there is no difference between them I moved the ftp url to a comment and changed src to fetchFromGitHub

Reply:
> I checked the repo and tarball with nix-diff, and since there is no difference between them

I must have messed up the invocation... there _is_ a difference: the release tarball is already reconfigured. I Had to add `autoreconfHook`.